tests/Unit/PointwiseFunctions/AnalyticSolutions/NewtonianEuler/Test_LaneEmdenStar.cpp

```diff
@@ -0,0 +1,98 @@
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <algorithm>
+#include <limits>
+#include <string>
+#include <tuple>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Evolution/Systems/NewtonianEuler/Tags.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/NewtonianEuler/LaneEmdenStar.hpp"
+#include "Utilities/MakeString.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+#include "tests/Unit/Pypp/CheckWithRandomValues.hpp"
+#include "tests/Unit/Pypp/SetupLocalPythonEnvironment.hpp"
+#include "tests/Unit/TestCreation.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+
+namespace {
+
+void test_construction_and_serialization() {
+  const double central_mass_density = 0.7;
+  const double polytropic_constant = 2.0;
+  NewtonianEuler::Solutions::LaneEmdenStar star(central_mass_density,
+                                                polytropic_constant);
+
+  const std::string input =
+      MakeString{} << "  CentralMassDensity: " << central_mass_density
+                   << "\n  PolytropicConstant: " << polytropic_constant;
+  const auto star_from_options =
+      TestHelpers::test_creation<NewtonianEuler::Solutions::LaneEmdenStar>(
+          input);
+
+  CHECK(star_from_options == star);
+
+  NewtonianEuler::Solutions::LaneEmdenStar star_to_move(central_mass_density,
+                                                        polytropic_constant);
+  test_move_semantics(std::move(star_to_move), star);  //  NOLINT
+
+  test_serialization(star);
+}
+
+struct LaneEmdenStarProxy : NewtonianEuler::Solutions::LaneEmdenStar {
+  using NewtonianEuler::Solutions::LaneEmdenStar::LaneEmdenStar;
+
+  template <typename DataType>
+  using variables_tags =
+      tmpl::list<NewtonianEuler::Tags::MassDensity<DataType>,
+                 NewtonianEuler::Tags::Velocity<DataType, 3, Frame::Inertial>,
+                 NewtonianEuler::Tags::SpecificInternalEnergy<DataType>,
+                 NewtonianEuler::Tags::Pressure<DataType>>;
+
+  template <typename DataType>
+  tuples::tagged_tuple_from_typelist<variables_tags<DataType>>
+  primitive_variables(const tnsr::I<DataType, 3, Frame::Inertial>& x,
+                      double t) const noexcept {
+    return this->variables(x, t, variables_tags<DataType>{});
+  }
+};
+
+template <typename DataType>
+void test_solution(const DataType& used_for_size,
+                   const double central_mass_density,
+                   const double polytropic_constant) noexcept {
+  LaneEmdenStarProxy star(central_mass_density, polytropic_constant);
+  pypp::check_with_random_values<
+      1, typename LaneEmdenStarProxy::template variables_tags<DataType>>(
+      &LaneEmdenStarProxy::template primitive_variables<DataType>, star,
+      "LaneEmdenStar",
+      {"mass_density", "velocity", "specific_internal_energy", "pressure"},
+      // with polytropic_constant == 2, star has outer radius ~ 1.77
+      {{{-2.0, 2.0}}},
```

Comment:
don't test with negative quantities that are only physically positive...

Reply:
I don't understand the change request. Here I am trying to sample (x,y,z) coords. Is my `pypp` invocation incorrect?

Reply:
never mind, my bad...